File: Submissions/#0032_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n, d;
  cin >> n >> d;

  vector<int> a(n);
  for (int i = 0; i < n; i++) {
    cin >> a[i];
  }

  int answer = 0;

  deque<int> que;
  for (int i = 0; i < n; i++) {
    que.clear();
    que.push_back(a[i]);
    for (int j = 0; j < n; j++) {
      if (i != j) {
        if (abs(a[i] - a[j]) <= d) {
          que.push_back(a[j]);
        }
      }
    }
    answer += que.size() - 1;
  }

  cout << answer << endl;

}

File: Submissions/#0032_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int d = sc.nextInt();

    int[] a = new int[n];
    for (int i = 0; i < n; i++) {
      a[i] = sc.nextInt();
    }
    sc.close();

    int answer = 0;

    Deque<Integer> que = new ArrayDeque<>();
    for (int i = 0; i < n; i++) {
      que.clear();
      que.addLast(a[i]);
      for (int j = 0; j < n; j++) {
        if (i != j) {
          if (Math.abs(a[i] - a[j]) <= d) {
            que.addLast(a[j]);
          }
        }
      }
      answer += que.size() - 1;
    }

    System.out.println(answer);

  }

}